pac4j-core/src/main/java/org/pac4j/core/client/Client.java

```diff
@@ -38,7 +40,7 @@
      * @param context the current web context
      * @return the performed redirection
      */
-    HttpAction redirect(WebContext context);
+    Optional<HttpAction> redirect(WebContext context);
```

Comment:
I don't think we should make this output `Optional`, functionally speaking, it doesn't make any sense.

And I'd like to asses the move of this method in the `IndirectClient` class.